app/services/request_password_reset.rb

```diff
@@ -1,16 +1,15 @@
 RequestPasswordReset = Struct.new(:email) do
   def perform
-    if user_not_found? || user_found_but_is_an_admin_or_tech?
-      # do nothing
-      #
-      # To prevent revealing account existence, we don't treat a
-      # "user not found" scenario as an error, unlike Devise's default
-      # behavior. That way, when the PasswordsController finishes processing
-      # the action, it will look like a successful transaction. We act as if
-      # the password reset email was sent, when in fact it wasn't.
-      # Similary, for security purposes, we don't want to allow password
-      # recovery via email for admin and tech support users.
-    elsif user.confirmed?
+    # To prevent revealing account existence, we don't treat a
+    # "user not found" scenario as an error, unlike Devise's default
+    # behavior. That way, when the PasswordsController finishes processing
+    # the action, it will look like a successful transaction. We act as if
+    # the password reset email was sent, when in fact it wasn't.
+    # Similary, for security purposes, we don't want to allow password
+    # recovery via email for admin and tech support users.
+    return if user_not_found? || user_found_but_is_an_admin_or_tech?
```

Comment:
I agree that a simple `if` statement should not need an early return, but in this case, the first part of the conditional wasn't doing anything, which I think is confusing, and not typical Ruby. 

If the user is not found, or is found but is an admin or tech, we don't want to do anything, and I think an early return explains that best. If neither of those conditions are true, then we want to do something.

We could also configure the GuardClause cop with a higher `MinBodyLength`, but that should probably go in a separate PR after a team discussion.